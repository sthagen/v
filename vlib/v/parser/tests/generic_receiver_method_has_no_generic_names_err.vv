struct Queue<T>{
	buffer []T
}

fn (q Queue<T>) poll() T {
	return q.buffer[0]
}

fn main() {
	q := Queue<int>{}
	println(q)
}
