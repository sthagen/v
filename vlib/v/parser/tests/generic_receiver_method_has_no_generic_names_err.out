vlib/v/parser/tests/generic_receiver_method_has_no_generic_names_err.vv:5:17: error: generic receiver method `poll` should add generic names, e.g. poll<T>
    3 | }
    4 |
    5 | fn (q Queue<T>) poll() T {
      |                 ~~~~
    6 |     return q.buffer[0]
    7 | }
